modules/client/src/main/java/org/apache/ignite/client/IgniteClient.java

```diff
@@ -387,6 +389,20 @@ public Builder sqlPartitionAwarenessMetadataCacheSize(int size) {
             return this;
         }
 
+        /**
+         * Sets the client name. Default is {@code null}, which means that Ignite will generate a unique name automatically.
+         *
+         * <p>Client name is used for identifying clients in JMX metrics.
+         *
+         * @param name Client name.
+         * @return This instance.
+         */
```

Comment:
Should we mention what happens on the name clash and whether the name should be unique?

Reply:
Good point, done (javadoc updated, test added).